src/reducers/bookmark.test.ts

```diff
@@ -0,0 +1,122 @@
+import {DEFAULT_QUERY_CONFIG} from 'compassql/build/src/config';
+import {SpecQueryModel} from 'compassql/build/src/model';
+import {SpecQuery} from 'compassql/build/src/query/spec';
+import {Schema} from 'compassql/build/src/schema';
+import {Channel} from 'vega-lite/build/src/channel';
+import {Mark} from 'vega-lite/build/src/mark';
+import {BOOKMARK_ADD_PLOT, BOOKMARK_MODIFY_NOTE, BOOKMARK_REMOVE_PLOT} from '../actions';
+import {Bookmark, BookmarkItem} from '../models';
+import {convertToPlotObjectsGroup, extractPlotObjects, PlotObject} from '../models/plot';
+import {bookmarkReducer} from './bookmark';
+
+
+
+describe('reducers/bookmark', () => {
+  const schema = new Schema({fields: []});
+  function buildSpecQueryModel(specQ: SpecQuery) {
+    return SpecQueryModel.build(specQ, schema, DEFAULT_QUERY_CONFIG);
+  }
+
+  function buildSpecQueryModelGroup(specQs: SpecQuery[]) {
+    const items = specQs.map(specQ => buildSpecQueryModel(specQ));
+    return {
+      name: 'a name',
+      path: 'path',
+      items: items,
+    };
+  }
+
+  const group = buildSpecQueryModelGroup([
+    {
+      mark: Mark.BAR,
+      encodings: [
+        {channel: Channel.X}
+      ]
+    }
+  ]);
+
+  const data = {url: 'a/data/set.csv'};
+  const plotObjectGroup = convertToPlotObjectsGroup(group, data);
+  const plotObject: PlotObject = extractPlotObjects(plotObjectGroup)[0];
```

Comment:
Why do you have to do this much setup for a PlotObject?  
Why don't you just declare one sensible plotObject directly?